enumerable.rb

```diff
@@ -0,0 +1,149 @@
+module Enumerable
+  def my_each
+    if block_given?
+      i = 0
+      while i < results.length
+        yield(results[i])
+        i += 1
+      end
+      results
+    else
+      results.to_enum
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      i = 0
+      while i < results.length
+        yield(results[i], i)
+        i += 1
+      end
+      results
+    else
+      results.to_enum
+    end
+  end
+
+  def my_select
+    if block_given?
+      choice = []
+      results.my_each do |x|
+        if yield(x)
+          choice.push(x)
+        end
+      end
+      choice
+    else
+      results.to_enum
+    end
+  end
+
+  def my_all?(items = nil)
+    if block_given?
+      my_each { |x| return false unless yield x }
+    else
+      return my_all? { |obj| obj } unless items
+
+      if items.class == Regexp
+        my_each { |x| return false unless items.match?(x) }
+      elsif items.class == Class
+        my_each { |x| return false unless x.is_a? items }
+      else
+        my_each { |x| return false unless x == items }
+      end
+    end
+    true
+  end
+
+  def my_any?(items = nil)
+    result = false
+    if block_given?
+      my_each { |item| result = true if yield item }
+    elsif items
+      my_each { |item| result = true if itemstern?(item, items) }
+    else
+      my_each { |item| result = true if item }
+    end
+    result
+  end
+
+  def my_none?(items = nil)
+    result = true
+    if block_given?
+      my_each { |item| result = false if yield item }
+    elsif items
+      my_each { |item| result = false if itemstern?(item, items) }
+    else
+      my_each { |item| result = false if item }
+    end
+    result
+  end
```

Comment:
- [ ] returns true if the block never returns true for all elements
- [ ] when no block or argument is given returns true only if none of the collection members is true
- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex
- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern